model/device.go

```diff
@@ -93,6 +93,136 @@ const (
+	"AlarmRecoveryVoltage": "840",   // 25.4545 V.
+	"AlarmVoltage":         "825",   // 25 V.
+	"AutoRestart":          "600",   // 10 minutes.
+	"Power":                "false", // initialise to OFF.
+	"Pulses":               "3",     // 3 pulses per cycle.
+	"PulseWidth":           "2",     // 1s ON, 1s OFF.
+	"PulseCycle":           "30",    // 30s between flashes.
+	"Suppress":             "false",
+}
+
+// controllerDefaultSensors contains a slice of the default sensors for a controller device.
+var controllerDefaultSensors = []SensorV2{
+	{Name: "Battery Voltage", Pin: "A0", Quantity: "DCV", Func: "scale", Args: "0.0289", Units: "V", Format: "round1"},
+	{Name: "Analog Value", Pin: "X10", Quantity: "OTH", Func: "none", Format: "round1"},
+	{Name: "Air Temperature", Pin: "X50", Quantity: "MWH", Func: "linear", Args: "0.1,-273.15", Units: "C", Format: "round1"},
+	{Name: "Humidity", Pin: "X51", Quantity: "MMB", Func: "scale", Args: "0.1", Units: "%", Format: "round2"},
+	{Name: "Water Temperature", Pin: "X60", Quantity: "MTW", Func: "linear", Args: "0.1,-273.15", Units: "C", Format: "round1"},
+}
+
+// controllerDefaultActs contains a slice of the default actuators for a controller device.
+var controllerDefaultActs = []ActuatorV2{
+	{Name: "Empty 1", Var: "Power1", Pin: "D16"},
+	{Name: "Empty 2", Var: "Power2", Pin: "D14"},
+	{Name: "Empty 3", Var: "Power3", Pin: "D15"},
+}
+
+// defaultVars contains a map between the device type and the default vars for that type.
+var defaultVars = map[string](map[string]string){
+	DevTypeController: controllerDefaultVars,
+}
+
+// defaultSensors contains a map between the device type and the default sensors for that type.
+var defaultSensors = map[string]([]SensorV2){
+	DevTypeController: controllerDefaultSensors,
+}
+
+// defaultActs contains a map between the device type and the default actuators for that type.
+var defaultActs = map[string]([]ActuatorV2){
+	DevTypeController: controllerDefaultActs,
+}
+
+// DeviceOption is a functional option to be supplied to NewDevice.
+type DeviceOption func(*Device) error
+
+// NewDevice creates a new device with the given parameters, and options.
+//
+// NOTE: Options are applied in order and any order dependent options should be
+// passed in order of execution.
+func NewDevice(ctx context.Context, skey int64, name, MAC string, settingsStore datastore.Store, options ...DeviceOption) (*Device, error) {
+	// Create an empty device.
+	dev := &Device{}
+
+	// Apply common config paramaters.
+	dev.Skey = skey
+	dev.Name = name
+	dev.Mac = MacEncode(MAC)
+	dev.ActPeriod = defaultActPeriod
+	dev.MonitorPeriod = defaultMonPeriod
+	dev.Enabled = true
+
+	// Apply the functional options.
+	var err error
+	for i, opt := range options {
+		err = opt(dev)
+		if err != nil {
+			return nil, fmt.Errorf("unable to apply option[%d]: %w", i, err)
+		}
+	}
+
+	switch dev.Type {
+	case DevTypeController:
+		dev.Inputs = defaultControllerInputs
```

Comment:
This also could be done with options, I.e, `WithInputs` and `WithOuputs`, and exporting the default inputs and outputs. There there is no device-specific code inside `NewDevice`.

For example:

```
devController, err = model.NewDevice(... model.WithInputs(model.DefaultControllerInputs) ...)

devCamera, err = model.NewDevice( ... model.WithInputs(model.DefaultCameraInputs) ...)

devHydrophone, err = model.NewDevice( ... model.WithInputs(model.DefaultHydrophoneInputs) ...)


```